homeassistant/components/trello/config_flow.py

```diff
@@ -0,0 +1,220 @@
+"""Config flow for Trello integration."""
+from typing import Any
+
+from trello import Member, TrelloClient, Unauthorized
+import voluptuous as vol
+from voluptuous.schema_builder import Schema
+
+from homeassistant import config_entries
+from homeassistant.const import CONF_API_KEY
+from homeassistant.core import callback
+from homeassistant.data_entry_flow import FlowResult
+from homeassistant.helpers import config_validation as cv, device_registry as dr
+
+from . import TrelloAdapter
+from .const import (
+    CONF_API_TOKEN,
+    CONF_OPTIONS_BOARDS,
+    CONF_USER_EMAIL,
+    CONF_USER_ID,
+    DOMAIN,
+    LOGGER,
+    USER_INPUT_BOARD_IDS,
+)
+
+CREDS_FORM_SCHEMA = vol.Schema(
+    {
+        vol.Required(CONF_API_KEY): str,
+        vol.Required(CONF_API_TOKEN): str,
+    }
+)
+
+
+class ConfigFlow(config_entries.ConfigFlow, domain=DOMAIN):
+    """Handle a config flow for the initial setup of a Trello integration."""
+
+    VERSION = 1
+
+    def __init__(self) -> None:
+        """Initialize config flow."""
+        self.api_key: str = ""
+        self.api_token: str = ""
+        self.user_email: str = ""
+        self.user_id: str = ""
+        self.ids_boards: dict[str, dict[str, str]] = {}
+        self.trello_adapter: TrelloAdapter
+
+    @staticmethod
+    @callback
+    def async_get_options_flow(
+        config_entry: config_entries.ConfigEntry,
+    ) -> config_entries.OptionsFlow:
+        """Create the options flow."""
+        return OptionsFlowHandler(config_entry)
```

Comment:
Perhaps move the options flow for a later PR

Reply:
That's a good point, should have thought of that. That will clean up a lot of code and tests. Think I might do that actually!